frontend/src/app/testQueue/AoEForm/AoEForm.tsx

```diff
@@ -0,0 +1,280 @@
+import React, { useState } from "react";
+import { gql } from "@apollo/client";
+import {
+  globalSymptomDefinitions,
+  getTestTypes,
+  getPregnancyResponses,
+} from "../../../patientApp/timeOfTest/constants";
+import RadioGroup from "../../commonComponents/RadioGroup";
+import Button from "../../commonComponents/Button";
+import FormGroup from "../../commonComponents/FormGroup";
+import { useQuery } from "@apollo/client";
+import "./AoEForm.scss";
+import SymptomInputs from "./SymptomInputs";
+import PriorTestInputs from "./PriorTestInputs";
+import { Redirect } from "react-router";
+
+interface Data {
+  patient: {
+    lastTest: {
+      dateTested: string;
+      result: string;
+    };
+  };
+}
+
+export const LAST_TEST_QUERY = gql`
+  query GetPatientsLastResult($patientId: String!) {
+    patient(id: $patientId) {
+      lastTest {
+        dateTested
+        result
+      }
+    }
+  }
+`;
+
+// Get the value associate with a button label
+// TODO: move to utility?
+const findValueForLabel = (
+  label: string,
+  list: { label: string; value: string }[]
+) => (list.filter((item) => item.label === label)[0] || {}).value;
+
+interface Props {
+  saveButtonText: string;
+  onClose?: () => void;
+  patient: {
+    internalId: string;
+    gender: string;
+  };
+  loadState?: {
+    noSymptoms: boolean;
+    symptoms: string;
+    symptomOnset: string;
+    priorTestDate: string;
+    priorTestResult: string;
+    priorTestType: string;
+    firstTest: boolean;
+    pregnancy: string;
+  };
+  saveCallback: (response: {
+    noSymptoms: boolean;
+    symptoms: string;
+    symptomOnset: string | undefined;
+    priorTestDate: string | undefined | null;
+    priorTestResult: string | undefined | null;
+    priorTestType: string | undefined | null;
+    firstTest: boolean;
+    pregnancy: string | undefined;
+  }) => void;
+  isModal: boolean;
+  noValidation: boolean;
+}
+
+const AoEForm: React.FC<Props> = ({
+  saveButtonText,
+  onClose,
+  patient,
+  loadState = {},
+  saveCallback,
+  isModal,
+  noValidation,
+}) => {
+  // this seems like it will do a bunch of wasted work on re-renders and non-renders,
+  // but it's all small-ball stuff for now
+  const testConfig = getTestTypes();
+  const symptomConfig = globalSymptomDefinitions;
+  const initialSymptoms: { [key: string]: boolean } = {};
+  if (loadState.symptoms) {
+    const loadedSymptoms: { [key: string]: string | boolean } = JSON.parse(
+      loadState.symptoms
+    );
+
+    symptomConfig.forEach((opt) => {
```

Comment:
This part isn't new but always seemed strange to me. We should know if we're dealing with strings or booleans.